osc-server/src/main/java/org/osc/core/broker/service/securitygroup/BindSecurityGroupService.java

```diff
@@ -100,24 +102,29 @@ public BaseJobResponse exec(BindSecurityGroupRequest request, EntityManager em)
                                     this.securityGroup.getName(), vs.getDistributedAppliance().getName()));
                 }
 
+                Set<Policy> policies = new HashSet<>();
                 Policy policy = null;
                 boolean isPolicyMappingSupported = this.apiFactoryService.syncsPolicyMapping(vs);
 
-                if (serviceToBindTo.getPolicyId() == null) {
-                    if (isPolicyMappingSupported) {
-                        throw new VmidcBrokerValidationException(
-                                "Service to bind: " + serviceToBindTo + " must have a policy id.");
-                    }
-                } else if (!isPolicyMappingSupported) {
-                    throw new VmidcBrokerValidationException(
-                            "Service to bind: " + serviceToBindTo + " is associated with a manager that does not support "
-                                    + "policy mapping and it should not have a policy id.");
-                } else {
-                    policy = PolicyEntityMgr.findById(em, serviceToBindTo.getPolicyId());
-
-                    if (policy == null) {
+                for(Long policyId : serviceToBindTo.getPolicyIds()){
```

Comment:
What if getPolicyIds() is empty or null? In that case if the sec mgr supports pol mapping the same exception below should throw but it seems in that case this loop will skip completely.

Reply:
Done. Please review the change.